httpx/middleware/newrelic_tracer.go

```diff
@@ -0,0 +1,59 @@
+package middleware
+
+import (
+	"fmt"
+	"github.com/remind101/nra"
+	"github.com/remind101/pkg/httpx"
+	"golang.org/x/net/context"
+	"net/http"
+)
+
+type NewRelicTracer struct {
+	handler  httpx.Handler
+	tracer   nra.TxTracer
+	router   *httpx.Router
+	createTx func(string, string, nra.TxTracer) nra.Tx
+}
+
+func NewRelicTracing(h httpx.Handler, router *httpx.Router, tracer nra.TxTracer) *NewRelicTracer {
+	return &NewRelicTracer{h, tracer, router, createTx}
+}
+
+func (h *NewRelicTracer) ServeHTTPContext(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+	templatePath := h.templatePath(r)
+	transactionName := fmt.Sprintf("%s (%s)", templatePath, r.Method)
+
+	tx := h.createTx(transactionName, r.URL.String(), h.tracer)
+	ctx = nra.WithTx(ctx, tx)
+
+	tx.Start()
+	defer tx.End()
+
+	return h.handler.ServeHTTPContext(ctx, w, r)
+}
+
+func (h *NewRelicTracer) templatePath(r *http.Request) string {
```

Comment:
Minor style thing, but in general, I think it's good to pull stuff like this into a func if it doesn't actually use anything from the struct and it's not there to satisfy an interface. Pulling it into a func makes it easier to test without having to setup a NewRelicTracer (if you wanted to unit test it).